File: src/my_project/interviews/top_150_questions_round_22/ex_11_h_index.py
from typing import List, Union, Collection, Mapping, Optional
from abc import ABC, abstractmethod

class Solution:
    def hIndex(self, citations: List[int]) -> int:

        citations.sort(reverse=True)
        len_cit:int = len(citations)
        answer:int = 0

        for i in range(len_cit):

            if citations[i] >= i + 1:
                answer = i + 1

        return answer

File: src/my_project/interviews_typescript/top_150_questions_round_23/ex_11_h_index.ts
function hIndex(citations: number[]): number {
    citations.sort((a, b) => b - a);
    const lenCit: number = citations.length;
    let answer: number = 0;

    for (let i = 0; i < lenCit; i++) {
        if (citations[i] >= i + 1) {
            answer = i + 1;
        }
    }

    return answer;
};        
